sc-memory/cpp/utils/sc_log.cpp

```diff
@@ -37,9 +42,33 @@ ScLog * ScLog::GetInstance()
 }
 
 ScLog::ScLog()
-  : m_mode(Type::Debug)
-  , m_isMuted(false)
 {
+  m_isMuted = false;
+  m_mode = Type::Info;
+  m_output_mode = OutputType::Console;
+
+  std::string s_mode = LOG_MODE;
+  int typeSize = sizeof(kTypeToStr) / sizeof(char *);
+  for (int i = 0; i < typeSize; i++)
+  {
+    std::string mode = kTypeToStr[i];
+    if (s_mode == mode)
+    {
+      m_mode = Type(i);
+    }
+  }
+
+  std::string s_output_mode = LOG_OUTPUT_TYPE;
+  int outputTypeSize = sizeof(kOutputTypeToStr) / sizeof(char *);
```

Comment:
Move this logic to separate method and use it for these cases(log type and log mode)